system-int-test/test/cmr/system_int_test/search/service/concept_retrieval_test.clj

```diff
@@ -0,0 +1,180 @@
+(ns cmr.system-int-test.search.service.concept-retrieval-test
+  "Integration test for service retrieval via the following endpoints:
+
+  * /concepts/:concept-id
+  * /concepts/:concept-id/:revision-id"
+  (:require
+   [cheshire.core :as json]
+   [clojure.test :refer :all]
+   [cmr.common.mime-types :as mt]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.service-util :as service]))
+
+(use-fixtures
+ :each
+ (join-fixtures
+  [(ingest/reset-fixture {"provguid1" "PROV1"})
+   service/grant-all-service-fixture]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Helper utility functions
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- get-svc
```

Comment:
Both this one and the next one were created (copied here; created in the variable concept retrieval test) because the `let` blocks that contained them were getting unruly: long, nested calls inside them, side-effect operations that returned no useful value (causing us to use the `[_ (call)]` anti-pattern), etc.

But I simultaneously harbour objections similar in nature to yours: does this mean that we then have little one-off functions scattered around our testing codebase whose sole purpose it to help us keep clean code? This would indicate that a symptom was addressed, not a root problem.

Which begs the question: what are we not taking into consideration with our APIs such that its usage is so awkward? A good API has very little of these usage issues; an excellent API has none of them.

In the mean time, I'll take another look and see if there's a nicer way to clean these up here...

Reply:
I find these helper functions make the test hard to follow. I don't want my `get-svc` call to actually ingest a service concept with the given service-name and returns the ingested service concept id and revision id. I would rather see something like `ingest-service-with-attrs` which at least tells me a service concept is ingested and that info is more important to understand the test.

Reply:
Yeah, that's a good point; those helper function names hide a benefit of the long names: self-documentation.  

We might be able to address my other concerns in the sys-int utility namespaces for now (and then, ultimately, a well-designed client library ... or utility library built on the client library).